Backoffice/AuthorizeEdition/TemplateStrategy.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace OpenOrchestra\Backoffice\AuthorizeEdition;
+
+use OpenOrchestra\Backoffice\NavigationPanel\Strategies\TreeTemplatePanelStrategy;
+use OpenOrchestra\ModelInterface\Model\TemplateInterface;
+use OpenOrchestra\ModelInterface\Repository\TemplateRepositoryInterface;
+use Symfony\Component\Security\Core\Authorization\AuthorizationCheckerInterface;
+
+/**
+ * Class TemplateStrategy
+ */
+class TemplateStrategy implements AuthorizeEditionInterface
+{
+    /**
+     * @var TemplateRepositoryInterface
+     */
+    protected $templateRepository;
+
+    /**
+     * @var AuthorizationCheckerInterface
+     */
+    protected $autorizationChecker;
+
+    /**
+     * @param TemplateRepositoryInterface   $templateRepository
+     * @param AuthorizationCheckerInterface $authorizationChecker
+     */
+    public function __construct(TemplateRepositoryInterface $templateRepository, AuthorizationCheckerInterface $authorizationChecker)
+    {
+        $this->templateRepository = $templateRepository;
+        $this->autorizationChecker = $authorizationChecker;
+    }
+
+    /**
+     * @param mixed $document
+     *
+     * @return bool
+     */
+    public function support($document)
+    {
+        return $document instanceof TemplateInterface;
+    }
+
+    /**
+     * @param TemplateInterface|mixed $document
+     *
+     * @return bool
+     */
+    public function isEditable($document)
+    {
+        $return = false;
+
+        if ($document instanceof TemplateInterface) {
```

Comment:
as the check is already done in the support method, you are sure that the document given to the isEditable method is an instance of TemplateInterface